app/hooks/use-logout.ts

```diff
@@ -11,23 +11,28 @@ const useLogout = () => {
   const client = useApolloClient()
   const { resetState } = usePersistentStateContext()
 
-  const logout = useCallback(async (): Promise<void> => {
-    try {
-      await client.cache.reset()
-      await AsyncStorage.multiRemove([BUILD_VERSION])
-      await KeyStoreWrapper.removeIsBiometricsEnabled()
-      await KeyStoreWrapper.removePin()
-      await KeyStoreWrapper.removePinAttempts()
+  const logout = useCallback(
+    async (stateToDefault = true): Promise<void> => {
+      try {
+        await client.cache.reset()
```

Comment:
I wonder if `client.cache.reset()` should be moved into a cleanup function in the useEffect that creates the client. My concern is that the cache could be reset in line 17 and then throughout the remainder of the function whenever the calls are awaited, there is a possibility for a rerender where the client cache would be repopulated again.